### DISCUSSION TITLE
Secrets and Environments

### DISCUSSION THREAD

**Discussion by bbaaxx** [FlowiseAI/Flowise#817] [Ideas]
I would like to propose the addition of secrets and environments. 
This feature will allow users to securely store and organize user values, including (but not restricted to) sensitive information such as API keys and credentials. 
Environments can be defined to manage different contexts, such as testing, staging, and production.

Examples can be found on software like Postman or [Hoppscotch](https://docs.hoppscotch.io/documentation/getting-started/rest/environment-variables)

Useful in cases like:
![image](https://github.com/FlowiseAI/Flowise/assets/3812307/e7e72d6c-1d75-4563-aefc-f761b95e15d0)

**Comment by HenryHengZJ** [comment]
thanks for the suggestion, we're thinking of allowing user to create custom credential and use it within the custom tool

**Reply by bbaaxx** [reply]
Thanks for your reply, it makes sense I guess is more "idiomatic" to provide such values from a node
